generate-weekly-changelog.sh

```diff
@@ -64,6 +64,8 @@ cat << EOF > /tmp/weekly.yaml
 # MALFORMED FILE CONTENTS WILL BREAK THE SITE BUILD
 EOF
 
+python3 -m venv venv
+source venv/bin/activate
```

Comment:
I think this should be outside of the script and should be in the workflow, people may run this locally when testing and setting up the environment shouldn't be part of the main script imo.